motors/tacho_motor_class.c

```diff
@@ -436,76 +629,69 @@ static ssize_t command_store(struct device *dev, struct device_attribute *attr,
 			     const char *buf, size_t size)
 {
 	struct tacho_motor_device *tm = to_tacho_motor(dev);
+	unsigned supported_commands;
 	int i, err;
-	bool start_timer = false;
-
-	for (i = 0; i < NUM_TM_COMMANDS; i++) {
-		if (sysfs_streq(buf, tacho_motor_command_names[i].name))
-			break;
-	}
 
-	if (i >= NUM_TM_COMMANDS)
-		return -EINVAL;
+	supported_commands = get_supported_commands(tm);
 
-	if (!(BIT(i) & get_supported_commands(tm)))
-		return -EINVAL;
+	for (i = 0; i < NUM_TM_COMMAND; i++) {
+		if (!sysfs_streq(buf, tacho_motor_command_names[i].name))
+			continue;
 
-	cancel_delayed_work_sync(&tm->run_timed_work);
-	if (i == TM_COMMAND_RUN_TIMED) {
-		i = TM_COMMAND_RUN_FOREVER;
-		start_timer = true;
-	}
-	if (i == TM_COMMAND_RESET)
-		tacho_motor_class_reset(tm);
```

Comment:
This should not be removed:

	if (i == TM_COMMAND_RESET)
		tacho_motor_class_reset(tm);